zipline/data/data_portal.py

```diff
@@ -527,20 +538,26 @@ def _get_history_daily_window(self, assets, end_dt, bar_count,
                                 index=days_for_window,
                                 columns=None)
 
-        data = []
+        future_data = []
+        eq_assets = []
 
         for asset in assets:
             if isinstance(asset, Future):
-                data.append(self._get_history_daily_window_future(
+                future_data.append(self._get_history_daily_window_future(
                     asset, days_for_window, end_dt, field_to_use
                 ))
             else:
-                data.append(self._get_history_daily_window_equity(
-                    asset, days_for_window, end_dt, field_to_use
-                ))
-
+                eq_assets.append(asset)
+        eq_data = self._get_history_daily_window_equities(
+            eq_assets, days_for_window, end_dt, field_to_use
+        )
+        if future_data:
```

Comment:
Not really part of your PR, but this made me realize we need to do a check to make sure all the assets passed to a `history` call are of the same type (this will become required in 24/5 futures backtesting).  Right now, if a future and an equity are passed in here, we'll only return the futures information.